sqlmesh/core/engine_adapter/base.py

```diff
@@ -366,22 +398,20 @@ def create_view(
             create_kwargs: Additional kwargs to pass into the Create expression
         """
         schema: t.Optional[exp.Table | exp.Schema] = exp.to_table(view_name)
-
-        if isinstance(query_or_df, DF_TYPES):
-            if PySparkDataFrame is not None and isinstance(query_or_df, PySparkDataFrame):
-                query_or_df = query_or_df.toPandas()
-
-            if not isinstance(query_or_df, pd.DataFrame):
-                raise SQLMeshError("Can only create views with pandas dataframes.")
-
+        df = self.try_get_pandas_df(query_or_df)
+        pyspark_df = self.try_get_pyspark_df(query_or_df)
+        if df is not None or pyspark_df:
```

Comment:
[Nit] We use `is not None` for `df` but not for `pyspark_df`.

Reply:
This is because of a nuance with pandas dataframes. They don't allow you to directly do boolean checks like that. It errors and provides options of types of checks you might want to do. So I do `is not None` just for pandas and then not for PySpark in order to make PySpark consistent with the rest of the codebase.